integr8ly/tutorial-web-app
# APPDUX-222: Add second phase of configuration management UI to Solution Explorer

## Motivation
JIRA:
https://issues.redhat.com/browse/APPDUX-222

Design spec:

## What
Add new sections of the config for writing new weekly maintenance window, choosing which maintenance window to upgrade (next or the one after next), and who to include on emails when an upgrade is available.

## Verification Steps

1. Log in with admin account.
2. Go to the Settings page.
3. Click the Managed Integration schedule tab.
4. Select a start time for backups and click Save.
5. Verify that if you go to settings again, the Next daily backup date/time has changed. Bonus points, go to the RHMIConfigs YAML on OpenShift and verify that the values have changed accordingly.
6. Verify that the Start time dropdown displays the current Next daily backup.
7. Verify that the weekly maintenance window date/time reflects what is in the YAML file.
8. Verify that the backup dropdown has disabled the time in which the weekly maintenance window occurs so that it is not selectable.
9. Select a new day and/or time for a weekly maintenance window. Verify that you can't pick a time that conflicts with the current next daily backup time. Click Save.
10. Verify that if you go to settings again, the Weekly maintenance window day/time has changed. Bonus points, go to the RHMIConfigs YAML on OpenShift and verify that the values have changed accordingly there as well.
11. Choose a radio button for whether to apply upgrades on the next maintenance day or the one following the next. Click Save. 
12. Verify that if you go to settings again, the radio button you chose is reflected. Bonus points, go to the RHMIConfigs YAML on OpenShift and verify that the values have changed accordingly there as well (in the config, a value of 0 for notBeforeDays is next, a value of 7 is following - waitForMaintenance is always true).
13. Enter one or more valid email addresses, separated by a comma. Click Save.
14. Verify that if you go into settings again, the email(s) you entered are reflected. Bonus points, go to the RHMIConfigs YAML on OpenShift and verify that the value for contacts has changed accordingly there as well.

## Checklist:

## Progress

## Additional Notes
**Testing:**
You can use my server for testing:
https://console-openshift-console.apps.cluster-uxddev-9f0b.uxddev-9f0b.example.opentlc.com/

Log in as customer-admin

Alternatively, you can load the following docker image onto your own OpenShift cluster:
docker.io/mfrances17/tutorial-web-app:APPDUX-222

**Screen caps:**

**Entire config:**

![config](https://user-images.githubusercontent.com/39063664/88598148-a7c8fd80-d036-11ea-886f-a6192115cee7.png)



**New functionality actual size:**

![config-actual-size](https://user-images.githubusercontent.com/39063664/88598158-ad264800-d036-11ea-8a70-04c8a6486f17.png)
